### PR TITLE
Add api end points

### PR DESCRIPTION
**Pull request by imubarak234**
Create an API endpoint to list all posts for a user.
Create an API endpoint to list all comments for a user's post.
Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
Add authentication to your API endpoints.
Your API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

**Comment by imubarak234** [comment]
Hello,

Please I know my commits are not in order, I ran into some complex git conflict issues and had to resolve them by merging to the development branch. 
 Thanks

File: spec/requests/posts_controller_spec.rb [hunk 1]
```diff
@@ -1,35 +1,35 @@
-# require 'rails_helper'
+require 'rails_helper'
```
Comment (andersonlebon) [inline spec/requests/posts_controller_spec.rb:2]: Your code in (app/controllers/api/v1/authentication_controller.rb)
```rb
module Api
  module V1
    class AuthenticationController < ApplicationController
      rescue_from ActionController::ParameterMissing, with: :parameter_missing

      def create
        params.require(:password)

        user = User.find_by(email: params.require(:email))
        token = AuthenticationTokenService.call(user.id)

        render json: { token: }, status: :created
      end

      private

      def parameter_missing(error)
        render json: { error: error.message }, status: :unprocessable_entity
      end
    end
  end
end
```

- [ ] Kindly make sure that your API `authenticate` endpoint is able to return an error when the user does not exit in the DB. 
- [ ] Also you are having a syntax error in the  API `authenticate` endpoint. 

image

Hint Example :point_down: 

```.rb
module Api
  module V1
    class AuthenticationController < ApplicationController
      rescue_from ActionController::ParameterMissing, with: :parameter_missing

      def create
        params.require(:password)

        user = User.find_by(email: params.require(:email))

        if user 
          token = AuthenticationTokenService.call(user.id)
          render json: { token: token }, status: :created
        else 
           render json: { error: "User does not exist" }
        end

      end

      private

      def parameter_missing(error)
        render json: { error: error.message }, status: :unprocessable_entity
      end
    end
  end
end
```

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @imubarak234 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DeliceLydia — APPROVED** [review]
## Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**